clients/nodejs/modules/JsonRpcServer.js

```diff
@@ -613,6 +642,27 @@ class JsonRpcServer {
         return block ? this._blockToObj(block, includeTransactions) : null;
     }
 
+    async streamBlocks(ws, query) {
```

Comment:
I believe you are currently restarting this method for every ws connection. This means you are also adding a new listener for the event(s) for each connection. Wouldn't it be better to have one broadcasting method that triggers for every head-change and automatically broadcasts to all sockets that registered for it?

Reply:
That's true. `/streamBlocks` is supposed to be used to _listen_ for new blocks over IPC, so I thought also using the `utils/Observable` listener in JS is the cleanest option. It also has fairly similar logic to the broadcaster method you described.

https://github.com/nimiq-network/core/blob/0cf652bb788942d14178d69853e0f9aee886bbd3/src/main/generic/utils/Observable.js#L47-L59

The listener is also removed as soon as the WS connection closes to keep everything clean.

What way would you prefer it?